content/fundamentals/execution-context.md

```diff
@@ -79,18 +112,18 @@ export interface ExecutionContext extends ArgumentsHost {
 }
 ```
 
-The `getHandler()` method returns a reference to the route handler about to be invoked. The `getClass()` method returns the type of the `Controller` class which this particular handler belongs to. For example, if the currently processed request is a `POST` request, destined for the `create()` method on the `CatsController`, `getHandler()` will return a reference to the `create()` method and `getClass()` will return a `CatsController` **type** (not instance).
+The `getHandler()` method returns a reference to the handler about to be invoked. The `getClass()` method returns the type of the `Controller` class which this particular handler belongs to. For example, in an HTTP context, if the currently processed request is a `POST` request, bound to the `create()` method on the `CatsController`, `getHandler()` returns a reference to the `create()` method and `getClass()` returns the `CatsController` **type** (not instance).
 
 ```typescript
 const methodKey = ctx.getHandler().name; // "create"
 const className = ctx.getClass().name; // "CatsController"
 ```
 
-Having an ability to access references to both class and method opens many doors for us. Most importantly, it gives us an opportunity to read the metadata set through the `@SetMetadata()` decorator from within guards or interceptors.
+The ability to access references to both the current class and handler method provides great flexibility. Most importantly, it gives us the opportunity to access the metadata set through the `@SetMetadata()` decorator from within guards or interceptors. We cover this use case below.
 
 #### Reflection and metadata
 
-Nest provides the ability to attach custom **metadata** to route handlers through the `@SetMetadata()` decorator. We can then access this metadata from within our class to make certain decisions.
+Nest provides the ability to attach **custom metadata** to route handlers through the `@SetMetadata()` decorator (read more about custom metadata [here](https://docs.nestjs.com/guards#reflection)). We can then access this metadata from within our class to make certain decisions.
```

Comment:
> (read more about custom metadata [here](https://docs.nestjs.com/guards#reflection)).

Actually, I think this chapter will provide somewhat more information about custom metadata than guards(?)